src/bpf/bpfctl.c

```diff
@@ -0,0 +1,220 @@
+/* SPDX-License-Identifier: LGPL-2.1-or-later */
+
+#include <getopt.h>
+
+#include "build.h"
+#include "format-table.h"
+#include "format-util.h"
+#include "main-func.h"
+#include "parse-argument.h"
+#include "pretty-print.h"
+#include "varlink.h"
+#include "verbs.h"
+
+static JsonFormatFlags arg_json_format_flags = JSON_FORMAT_OFF;
+
+struct bpf_prog_data {
+        char *name;
+        uint64_t type;
+        char *memlock;
+        char *tag;
+        char *map_names;
+        uint64_t run_time_ns;
+        uint64_t run_cnt;
+};
+
+struct bpf_map_data {
+        char *name;
+        uint64_t type;
+        char *memlock;
+};
+
+static int query_bpf_progs_and_maps_reply(
+                Varlink *link,
+                JsonVariant *parameters,
+                const char *error_id,
+                VarlinkReplyFlags flags,
+                void *userdata) {
+
+        int r;
+        JsonVariant *m, *v;
+
```

Comment:
btw, the `flags` param will tell you when you are done, i.e. if the VARLINK_REPLY_CONTINUE bit is unset this was the last reply

Reply:
Same as above